gst-plugin/src/image.cpp

```diff
@@ -200,24 +200,15 @@ uint8_t Image::save_bmp(const std::string& filename) const
 ******************************************/
 void Image::write_char(char code, int32_t x, int32_t y, int32_t color, int32_t backcolor)
 {
-    int32_t height;
-    int32_t width;
-    int8_t * p_pattern;
-    uint8_t mask = 0x80;
-
-    if ((code >= 0x20) && (code <= 0x7e))
-    {
-        p_pattern = (int8_t *)&g_ascii_table[code - 0x20][0];
-    }
-    else
-    {
-        p_pattern = (int8_t *)&g_ascii_table[10][0]; /* '*' */
-    }
+    auto& p_pattern = ((code >= 0x20) && (code <= 0x7e)) ?
+                                        g_ascii_table[code - 0x20]:
+                                        g_ascii_table[10]; /* '*' */
```

Comment:
hardcoded array entry 10? Why? Use a descriptive MACRO name instead, also what if you `g_ascii_table` doesn't have a sufficient number of indices? Check & fail gracefully